### PR TITLE
Add macros for retrieving file base name

### PR DESCRIPTION
**Pull request by vzhukovs**
### What does this PR do?
This PR proposal adds two new macros for retrieving _base_ name from the file in editor or project explorer. _Base_ - means _the name without extension_.

* `${explorer.current.file.basename}`
* `${editor.current.file.basename}`

eclipse che - wksp-ucrs 2017-03-27 16-11-56

eclipse che - wksp-ucrs 2017-03-27 16-11-11

Signed-off-by: Vladyslav Zhukovskii <vzhukovskii@codenvy.com>

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/4552

#### Changelog
Added macros for retrieving file base name

#### Release Notes
Added macros for retrieving file base name

#### Docs PR
https://github.com/eclipse/che-docs/pull/178

### CODE REVIEW

**Review by slemeur — APPROVED** [review]

**Review by slemeur — APPROVED** [review]

File: ide/che-core-ide-app/src/test/java/org/eclipse/che/ide/editor/macro/EditorCurrentFileBaseNameMacroTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    @Test
+    public void getEmptyValue() throws Exception {
+        provider.expand();
+
+        verify(editorAgent).getActiveEditor();
+        verify(promiseProvider).resolve(eq(""));
+    }
+
+}
```
Comment (skabashnyuk) [inline ide/che-core-ide-app/src/test/java/org/eclipse/che/ide/editor/macro/EditorCurrentFileBaseNameMacroTest.java]: new line

File: ide/che-core-ide-app/src/test/java/org/eclipse/che/ide/part/explorer/project/macro/ExplorerCurrentFileBaseNameMacroTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    @Test
+    public void getEmptyValues() throws Exception {
+        initWithNoFiles();
+
+        provider.expand();
+
+        verify(promiseProvider).resolve(eq(""));
+    }
+
+}
```
Comment (skabashnyuk) [inline ide/che-core-ide-app/src/test/java/org/eclipse/che/ide/part/explorer/project/macro/ExplorerCurrentFileBaseNameMacroTest.java]: new line

Comment (vzhukovs) [reply ide/che-core-ide-app/src/test/java/org/eclipse/che/ide/part/explorer/project/macro/ExplorerCurrentFileBaseNameMacroTest.java]: done, thx

File: ide/che-core-ide-app/src/test/java/org/eclipse/che/ide/editor/macro/EditorCurrentFileBaseNameMacroTest.java (hunk 1 above)
Comment (vzhukovs) [reply ide/che-core-ide-app/src/test/java/org/eclipse/che/ide/editor/macro/EditorCurrentFileBaseNameMacroTest.java]: done, thx

**Review by vparfonov — APPROVED** [review]

**Pull request merged**